modules/module-mongodb-storage/test/src/connection-report-storage.test.ts

```diff
@@ -0,0 +1,256 @@
+import { afterAll, beforeAll, describe, expect, it } from 'vitest';
+import { INITIALIZED_MONGO_REPORT_STORAGE_FACTORY } from './util.js';
+import { event_types } from '@powersync/service-types';
+
+function removeVolatileFields(
+  sdks: event_types.ClientConnection[]
+): Partial<event_types.ClientConnection & { _id: string }>[] {
+  return sdks.map((sdk: Partial<event_types.ClientConnection & { _id: string }>) => {
+    const { _id, disconnected_at, connected_at, jwt_exp, ...rest } = sdk;
+    return {
+      ...rest
+    };
+  });
+}
+
+describe('SDK reporting storage', async () => {
```

Comment:
Tests should also specifically include a case where `client_id` is undefined.

Reply:
If client_id is undefined, I have made it an empty string. It is added as a user connection in the tests.